### PR TITLE
Add endpoint for a POST request to `/api/v1/playlists`

### PR DESCRIPTION
**Pull request by BabsLabs**
Features of PR:
- Move favorties tests into a `/test/favorites` subdirectory.
- Make `/test/playlists` subdirectory.
- Add tests for a `POST` request to `/api/v1/playlists`.
- Add playlists route and router to `app.js`.
- Add endpoint for `POST` requests to `/api/v1/playlists` that adds a playlist to the playlists table.

Testing:
- Test coverage is 93.33%

Refactors / Thoughts:
- How do we check for created_at and updated_at?

### CODE REVIEW

File: routes/api/v1/playlists.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+const configuration = require('../../../knexfile')[environment];
+const database = require('knex')(configuration);
+
+router.post('/', async (request, response) => {
+  let playlistTitle = request.body.title;
+
+  if (playlistTitle) {
+
+    let playlist = { title: playlistTitle };
+    let existsResObj = await alreadyFavorite(playlist);
```
Comment (johnktravers) [inline routes/api/v1/playlists.js]: We might want to rename this method to alreadyPlaylist

File: routes/api/v1/playlists.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  } else {
+    return new ResponseObj(200, 'Playlist has not yet been favorited.');
+  }
+};
+
+async function addPlaylistToDB(playlist) {
+  let savedPlaylist = await database('playlists').insert(
+      {
+        title: playlist.title,
+      }, ['id', 'title', 'created_at', 'updated_at']);
```
Comment (johnktravers) [inline routes/api/v1/playlists.js]: Good work returning all of these fields

File: routes/api/v1/playlists.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    return new ResponseObj(500, 'New playlist cannot be saved into database. Please try again.');
+  }
+};
+
+function errorResponse(resObj, response) {
+  return response.status(resObj.status).json({
+    status: resObj.status,
+    errorMessage: resObj.payload
+  });
+};
```
Comment (johnktravers) [inline routes/api/v1/playlists.js]: We could factor this out into a module since it's used by both the favorites and playlists routers

File: tests/favorites/add-favorite.spec.js [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
 const environment = process.env.NODE_ENV || 'test';
-const configuration = require('../knexfile')[environment];
+const configuration = require('../../knexfile')[environment];
 const database = require('knex')(configuration);
 
 describe('Add favorites endpoint', () => {
   beforeEach(async () => {
-    await database.raw('truncate table favorites cascade');
+    await database.raw('TRUNCATE TABLE favorites RESTART IDENTITY CASCADE');
```
Comment (johnktravers) [inline tests/favorites/add-favorite.spec.js]: Good work finding this SQL command!

File: tests/playlists/add-playlist.spec.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+      .post("/api/v1/playlists")
+      .send({ title: "CodeSongs" })
+      .type('form');
+
+    expect(res.statusCode).toBe(409);
+    expect(res.body).toHaveProperty('status');
+    expect(res.body).toHaveProperty('errorMessage');
+    expect(res.body.status).toBe(409);
+    expect(res.body.errorMessage).toBe('A playlist with that title has already been added to your playlists!');
+  });
```
Comment (johnktravers) [inline tests/playlists/add-playlist.spec.js]: Nice test for that edge case!
